src/main/java/games/strategy/triplea/delegate/MoveValidator.java

```diff
@@ -133,14 +133,14 @@ public static MoveValidationResult validateMove(final Collection<Unit> units, fi
   static MoveValidationResult validateFirst(final GameData data, final Collection<Unit> units, final Route route,
       final PlayerID player, final MoveValidationResult result) {
     if (!units.isEmpty()
-        && !getEditMode(data)
-        && !Match.allMatchNotEmpty(Match.getMatches(units,
-            Matches.unitIsBeingTransportedByOrIsDependentOfSomeUnitInThisList(units, route, player, data, true)
-                .invert()),
-            Matches.unitIsOwnedBy(player))) {
-      result.setError("Player, " + player.getName() + ", is not owner of all the units: "
-          + MyFormatter.unitsToTextNoOwner(units));
-      return result;
+        && !getEditMode(data)) {
+      final Collection<Unit> matches = Match.getMatches(units,
+          Matches.unitIsBeingTransportedByOrIsDependentOfSomeUnitInThisList(units, route, player, data, true).invert());
+      if (matches.isEmpty() || !Match.allMatch(matches, Matches.unitIsOwnedBy(player))) {
+        result.setError("Player, " + player.getName() + ", is not owner of all the units: "
+            + MyFormatter.unitsToTextNoOwner(units));
+        return result;
+      }
```

Comment:
Can someone double/triple check this chunk. Its kind of hard to follow. Might be worth refactoring this a little to clean it up.